bioblend/galaxy/histories/__init__.py

```diff
@@ -76,26 +106,45 @@ def get_histories(self, history_id=None, name=None, deleted=False, published=Non
         :type published: bool or None
         :param published: whether to filter for the published histories
           (``True``) or for the non-published ones (``False``). If not set, no
-          filtering is applied.
+          filtering is applied. Note the filtering is only applied to the user's
+          own histories; to access all histories published by any user, use the
+          ``get_published_histories`` method.
+
+        :type slug: str
+        :param slug: History slug to filter on
 
         :rtype: list
         :return: Return a list of history element dicts. If more than one
                  history matches the given ``name``, return the list of all the
                  histories with the given name
         """
-        if history_id is not None and name is not None:
-            raise ValueError('Provide only one argument between name or history_id, but not both')
-        params = {}
-        if published is not None:
-            params.setdefault('q', []).append('published')
-            params.setdefault('qv', []).append(published)
-        histories = self._get(deleted=deleted, params=params)
-        if history_id is not None:
-            history = next((_ for _ in histories if _['id'] == history_id), None)
-            histories = [history] if history is not None else []
-        elif name is not None:
-            histories = [_ for _ in histories if _['name'] == name]
-        return histories
+        return self._get_histories(history_id=history_id, name=name, deleted=deleted, filter_user_published=published, get_all_published=False, slug=slug)
+
+    def get_published_histories(self, history_id=None, name=None, deleted=False, slug=None):
```

Comment:
I think filtering by `history_id` in a `get_*()` method is an anti-pattern, using the `show_*()` method is much more efficient, but unfortunately we have it `get_histories()` since the beginning.
I'd prefer not to add it to `get_published_histories()` though, if that's OK.